### PR TITLE
JBR-7420 Vulkan: Implement DRAW_PARALLELOGRAM primitive for flat color rendering

### PR DESCRIPTION
**Pull request by avu**
Refactored rendering code. Provided common implementation for fill and draw operations.

### CODE REVIEW

File: src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.c [hunk 1]
```diff
@@ -721,13 +724,25 @@ void VKRenderer_TextureRender(VKLogicalDevice* logicalDevice, VKImage *destImage
...
+void VKRenderer_ColorRender(VKLogicalDevice* logicalDevice, VKImage *destImage, VKRenderer *renderer, uint32_t rgba,
+                            VkBuffer vertexBuffer,
+                            uint32_t vertexNum)
 {
+    if (!vertexBuffer) {
+        J2dRlsTrace(J2D_TRACE_ERROR, "VKRenderer_ColorRender: vertex buffer is NULL\n")
+        return;
+    }
+    logicalDevice->vkWaitForFences(logicalDevice->device, 1, &logicalDevice->inFlightFence, VK_TRUE, UINT64_MAX);
+    logicalDevice->vkResetFences(logicalDevice->device, 1, &logicalDevice->inFlightFence);
```
Comment (bourgesl) [inline src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.c:736]: Maybe a macro or simple method to handle fences would help ?

**Review by bourgesl — APPROVED** [review]
Lgtm

File: src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.h [hunk 2]
```diff
@@ -40,28 +40,38 @@ struct VKRenderer {
     VkDescriptorSet     descriptorSets;
     VkPipelineLayout    pipelineLayout;
     VkPipeline          graphicsPipeline;
+    VkPrimitiveTopology primitiveTopology;
```
Comment (YaaZ) [inline src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.h:43]: Do we actually need this field?

**Review by YaaZ — APPROVED** [review]

File: src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.c (hunk 1 above)
Comment (avu) [reply src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.c:736]: Yes, I thought about it. I'll introduce something like this in upcoming fixes.

File: src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.h (hunk 2 above)
Comment (avu) [reply src/java.desktop/share/native/common/java2d/vulkan/VKRenderer.h:43]: It's used to provide the correct amount of points for the Render command (for draw operation you don't need the complete triangle.

**Pull request merged**